### PR TITLE
[Merged by Bors] - add more multiplexing_test

### PR DESCRIPTION
**Pull request by TanNgocDo**
This is an attempt to add more integration test for #3174 . 
Need to be update more case + fuzz test

### CODE REVIEW

File: crates/fluvio-socket/src/multiplexing.rs [hunk 1]
```diff
@@ -755,18 +773,112 @@ mod tests {
         assert_eq!(response.msg, "hello");
     }
 
+    async fn test_client_closed_socket<C: ConnectorHandler + 'static>(addr: &str, mut handler: C) {
+        use std::time::SystemTime;
+
+        sleep(Duration::from_millis(20)).await;
```
Comment (sehz) [inline crates/fluvio-socket/src/multiplexing.rs:779]: make this constant

**Comment by sehz** [comment]
clippy failures

**Comment by sehz** [comment]
@galibey @morenol Please review

File: crates/fluvio-socket/src/multiplexing.rs (hunk 1 above)
Comment (TanNgocDo) [reply crates/fluvio-socket/src/multiplexing.rs]: Updated it

File: crates/fluvio-socket/src/multiplexing.rs [hunk 2]
```diff
@@ -578,7 +583,21 @@ mod tests {
         }
     }
 
-    async fn test_server<A: AcceptorHandler + 'static>(addr: &str, mut handler: A) {
+    fn get_error_kind<T: std::fmt::Debug>(
+        result: Result<T, SocketError>,
+    ) -> Option<std::io::ErrorKind> {
+        match result.unwrap_err() {
+            SocketError::Io { source, .. } => Some(source.kind()),
```
[the remark this answers is not in the document]
Comment (TanNgocDo) [reply crates/fluvio-socket/src/multiplexing.rs]: thanks, updated it

File: crates/fluvio-socket/src/multiplexing.rs [hunk 3]
```diff
@@ -755,18 +775,112 @@ mod tests {
...
+    async fn test_client_time_out<C: ConnectorHandler + 'static>(addr: &str, mut handler: C) {
+        sleep(Duration::from_millis(SLEEP_MS * 2)).await;
+        debug!("client: trying to connect");
+        let tcp_stream = TcpStream::connect(&addr).await.expect("connection fail");
+        let socket = handler.connect(tcp_stream).await;
+        debug!("client: connected to test server and waiting...");
+        sleep(Duration::from_millis(SLEEP_MS * 2)).await;
+        let multiplexer: std::sync::Arc<MultiplexerSocket> = MultiplexerSocket::shared(socket);
+
+        let fut = async move {
```
[the remark this answers is not in the document]
Comment (TanNgocDo) [reply crates/fluvio-socket/src/multiplexing.rs]: updated

**Comment by galibey** [comment]
Need to increment patch version here https://github.com/infinyon/fluvio/blob/0377bae0ae3e1b32bb3e4b823bdfbb1f2e88a5a5/crates/fluvio-socket/Cargo.toml#L3 to make CI happy.

Probably, need to rebase from master as well.

**Comment by TanNgocDo** [comment]
> version = "0.14.2"

@galibey , should it be: "0.14.3", right ?

**Comment by galibey** [comment]
> > version = "0.14.2"
> 
> @galibey , should it be: "0.14.3", right ?

yes

**Review by galibey — APPROVED** [review]
LGTM

**Pull request closed without merging**